tempodb/encoding/vparquet3/schema.go

```diff
@@ -0,0 +1,733 @@
+	StartTimeUnixNano      uint64      `parquet:",delta"`
+	DurationNano           uint64      `parquet:",delta"`
+	StatusCode             int         `parquet:",delta"`
+	StatusMessage          string      `parquet:",snappy"`
+	Attrs                  []Attribute `parquet:",list"`
+	DroppedAttributesCount int32       `parquet:",snappy"`
+	Events                 []Event     `parquet:",list"`
+	DroppedEventsCount     int32       `parquet:",snappy"`
+	Links                  []byte      `parquet:",snappy"` // proto encoded []*v1_trace.Span_Link
+	DroppedLinksCount      int32       `parquet:",snappy"`
+
+	// Static dedicated attribute columns
+	HttpMethod     *string `parquet:",snappy,optional,dict"`
+	HttpUrl        *string `parquet:",snappy,optional,dict"`
+	HttpStatusCode *int64  `parquet:",snappy,optional"`
+
+	// Dynamically assignable dedicated attribute columns
+	DedicatedAttributes DedicatedAttributes `parquet:""`
+}
+
+func (s *Span) IsRoot() bool {
+	return len(s.ParentSpanID) == 0
+}
+
+type InstrumentationScope struct {
+	Name    string `parquet:",snappy,dict"`
+	Version string `parquet:",snappy,dict"`
+}
+
+type ScopeSpans struct {
+	Scope InstrumentationScope `parquet:""`
+	Spans []Span               `parquet:",list"`
+}
+
+type Resource struct {
+	Attrs []Attribute `parquet:",list"`
+
+	// Always empty for testing
+	Test string `parquet:",snappy,dict,optional"`
+
+	// Static dedicated attribute columns
+	ServiceName      string  `parquet:",snappy,dict"`
+	Cluster          *string `parquet:",snappy,optional,dict"`
+	Namespace        *string `parquet:",snappy,optional,dict"`
+	Pod              *string `parquet:",snappy,optional,dict"`
+	Container        *string `parquet:",snappy,optional,dict"`
+	K8sClusterName   *string `parquet:",snappy,optional,dict"`
+	K8sNamespaceName *string `parquet:",snappy,optional,dict"`
+	K8sPodName       *string `parquet:",snappy,optional,dict"`
+	K8sContainerName *string `parquet:",snappy,optional,dict"`
+
+	// Dynamically assignable dedicated attribute columns
+	DedicatedAttributes DedicatedAttributes `parquet:""`
+}
+
+type ResourceSpans struct {
+	Resource   Resource     `parquet:""`
+	ScopeSpans []ScopeSpans `parquet:"ss,list"`
+}
+
+type Trace struct {
+	// TraceID is a byte slice as it helps maintain the sort order of traces within a parquet file
+	TraceID []byte `parquet:""`
+	// TraceIDText is for better usability on downstream systems i.e: something other than Tempo is reading these files.
+	// It will not be used as the primary traceID field within Tempo and is only helpful for debugging purposes.
+	TraceIDText string `parquet:",snappy"`
+
+	// Trace-level attributes for searching
+	StartTimeUnixNano uint64 `parquet:",delta"`
+	EndTimeUnixNano   uint64 `parquet:",delta"`
+	DurationNano      uint64 `parquet:",delta"`
+	RootServiceName   string `parquet:",dict"`
+	RootSpanName      string `parquet:",dict"`
+
+	ResourceSpans []ResourceSpans `parquet:"rs,list"`
```

Comment:
Reordering of columns is technically a schema change because the column indexes change. For vParquet3 this isn't relevant as we already have schema changes.

It has no effect on performance or file size.